Jenkinsfile-publish-image

```diff
@@ -83,32 +83,55 @@ pipeline {
             }
         }
 
-        stage('Start instance') {
-            steps {
-                dir("dhis-2/dhis-e2e-test") {
-                    sh "IMAGE_NAME=${DOCKER_CORE_IMAGE_FULL_NAME} docker-compose -p ${COMPOSE_PARAMETER} up -d"
-                }
-            }
-        }
-
         stage('Run api tests') {
-            steps {
-                script {
-                    TESTS_IMAGE_FULL_NAME = "core-api-tests:${DOCKER_IMAGE_TAG}"
-
-                    dir("dhis-2/dhis-e2e-test") {
-                        sh "docker build -t ${TESTS_IMAGE_FULL_NAME} ."
-                        sh "IMAGE_NAME=${TESTS_IMAGE_FULL_NAME} docker-compose -p ${COMPOSE_PARAMETER} -f docker-compose.e2e.yml up --exit-code-from e2e-test"
+            matrix {
+                axes {
+                    axis {
+                        name 'DOCKER_VARIANT'
+                        values '-tomcat-8.5-jdk8-openjdk-slim', '-tomcat-9.0.27-jdk11-openjdk-slim'
```

Comment:
Not sure if we can pull it off, but it would be nice if we have a single place where we source the variants we build for both this file, and the containers defined in: https://github.com/dhis2/dhis2-core/blob/master/docker/shared/containers-list.sh#L26-L42

Making sure our container versions are up-to-date is currently something that we don't track super-well, and we are opening ourselves up for more complications when we hardcode them in multiple places.